src/main/java/org/opensearch/sdk/rest/ExtensionRestPathRegistry.java

```diff
@@ -57,20 +65,25 @@ public void registerHandler(ExtensionRestHandler restHandler) {
      * Registers a REST handler to be executed when one of the provided methods and path match the request.
      *
      * @param path Path to handle (e.g., "/{index}/{type}/_bulk")
-     * @param handler The handler to actually execute
+     * @param extensionRestHandler The handler to actually execute
+     * @param name The name corresponding to this handler
      * @param method GET, POST, etc.
      */
-    private void registerHandler(Method method, String path, ExtensionRestHandler extensionRestHandler) {
+    public void registerHandler(Method method, String path, Optional<String> name, ExtensionRestHandler extensionRestHandler) {
         pathTrie.insertOrUpdate(
             path,
             new SDKMethodHandlers(path, extensionRestHandler, method),
             (mHandlers, newMHandler) -> mHandlers.addMethods(extensionRestHandler, method)
         );
         if (extensionRestHandler instanceof ExtensionDeprecationRestHandler) {
-            registeredDeprecatedPaths.add(restPathToString(method, path));
+            registeredDeprecatedPaths.add(restPathToString(method, path, name));
             registeredDeprecatedPaths.add(((ExtensionDeprecationRestHandler) extensionRestHandler).getDeprecationMessage());
         } else {
-            registeredPaths.add(restPathToString(method, path));
+            if (name.isPresent()) {
+                registeredPaths.add(restPathToString(method, path, name));
+            } else {
+                registeredPaths.add(restPathToString(method, path, name));
```

Comment:
The method called in `else` is exactly the same as the method called in the `isPresent()` case. Why does this conditional exist?

Reply:
Removed this conditional